shielded-pool/src/component.rs

```diff
@@ -360,7 +388,11 @@ impl ShieldedPool {
             .await;
         // and the note commitment tree data and anchor:
         self.state
-            .set_nct_anchor(self.compact_block.height, self.note_commitment_tree.root2())
+            .set_nct_anchor(
+                self.compact_block.height,
+                self.note_commitment_tree.root2(),
+                self.tiered_commitment_tree.root(),
```

Comment:
This is writing the NCT anchor from the TCT sidecar into the consensus-critical state, which is what we don't want to do yet, right?

Reply:
Ah, yeah, I figured it was fine to prepone doing that, but taking it out for now.